app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersAssetsStorage.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.wallpapers
+
+import android.content.Context
+import android.content.res.AssetManager
+import org.json.JSONArray
+import org.json.JSONObject
+
+class WallpapersAssetsStorage(private val context: Context) : WallpaperStorage {
+
+    override fun all(): List<Wallpaper> {
+        val assetsManager = context.assets
+        return assetsManager.readArray("$WALLPAPERS_DIRECTORY/wallpapers.json").toWallpapers()
+    }
+
+    private fun JSONArray.toWallpapers(): List<Wallpaper> {
+        return (0 until this.length()).map { index ->
+            this.getJSONObject(index).toWallpaper()
+        }
+    }
+
+    private fun JSONObject.toWallpaper(): Wallpaper {
+        return Wallpaper(
+            name = getString("name"),
+            portraitPath = getString("portrait"),
+            landscapePath = getString("landscape"),
+            isDark = getBoolean("isDark")
+        )
+    }
+
+    private fun AssetManager.readArray(fileName: String) = JSONArray(
+        open(fileName).bufferedReader().use {
+            it.readText()
+        }
+    )
+
+    companion object {
+        private const val WALLPAPERS_DIRECTORY = "wallpapers"
+    }
```

Comment:
We could avoid an extra object instantiation by moving this inline, or as a private property on `WallpapersAssetsStorage`.